consensus/cryptarchia-engine/src/time.rs

```diff
@@ -57,3 +75,71 @@ impl Add<u32> for Epoch {
         Epoch(self.0 + rhs)
     }
 }
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct EpochConfig {
+    // The stake distribution is always taken at the beginning of the previous epoch.
+    // This parameters controls how many slots to wait for it to be stabilized
+    // The value is computed as epoch_stake_distribution_stabilization * int(floor(k / f))
+    pub epoch_stake_distribution_stabilization: u8,
+    // This parameter controls how many slots we wait after the stake distribution
+    // snapshot has stabilized to take the nonce snapshot.
+    pub epoch_period_nonce_buffer: u8,
+    // This parameter controls how many slots we wait for the nonce snapshot to be considered
+    // stabilized
+    pub epoch_period_nonce_stabilization: u8,
+}
+
+impl EpochConfig {
+    pub fn epoch_length(&self, base_period_length: u64) -> u64 {
+        (self.epoch_stake_distribution_stabilization as u64
```

Comment:
This should be replaced with checked variants, so that we don't risk overflows in release builds, which don't panic in those cases.

Reply:
There's also a clippy lints that checks for this, maybe we want to introduce it at some point...

Reply:
> There's also a clippy lints that checks for this, maybe we want to introduce it at some point...

Yes, we want. Anything that helps catch things earlier the better.

Reply:
> This should be replaced with checked variants, so that we don't risk overflows in release builds, which don't panic in those cases.

I added `NonZero` wherever possible as well. As it makes sense. It should force values from config to be valid as well. Thanks!

Reply:
@ntn-x2 , I have some changes that will address the limits from config so this operations should be safe. But it may grow this too much and go out of scope. I think I would add it later in a further PR.